### PR TITLE
feat: ActionPad. Add same expand/collapse behavior as ActionBar

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #656

## Summary

feat: ActionPad. Add same expand/collapse behavior as ActionBar #656

### CODE REVIEW

**Comment by driskull** [comment]
@asangma this needs a little bit of styling. The bottom action group needs a border to separate it from the other groups.

File: src/components/utils/CalciteExpandToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  layout: CalciteLayout;
+  toggleExpand: () => void;
+}
+
+const ICONS = {
+  chevronsLeft: "chevrons-left",
+  chevronsRight: "chevrons-right"
+};
+
+function getClosestShellLayout(el: HTMLElement): CalciteLayout {
```
Comment (jcfranco) [inline src/components/utils/CalciteExpandToggle.tsx]: Should this be` CalciteLayout | null` to make it explicit that null may be returned?

File: src/components/utils/CalciteExpandToggle.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  const shellNode = el.closest("calcite-shell-panel");
+
+  if (!shellNode) {
+    return;
+  }
+
+  return shellNode.layout;
+}
+
+export function setActionTextEnabled({
```
Comment (jcfranco) [inline src/components/utils/CalciteExpandToggle.tsx]: Suggestion: API-wise, what do you think about renaming this to `toggleChildActionText({ parent, enabled})` or something similar?

File: src/components/calcite-action-pad/calcite-action-pad.e2e.ts [hunk 3]
```diff
@@ -1,18 +1,108 @@
...
+  it("defaults", async () => {
+    const page = await newE2EPage();
+
+    await page.setContent("<calcite-action-pad></calcite-action-pad>");
+    const element = await page.find("calcite-action-pad");
+    expect(element.getAttribute("expand")).not.toBeNull();
+    expect(element.getAttribute("expanded")).toBeNull();
+  });
+
+  it("expand: true", async () => {
```
Comment (jcfranco) [inline src/components/calcite-action-pad/calcite-action-pad.e2e.ts]: Nitpick: expand tests could be nested in a `describe` or handled in a single test (reasoning: tests the same behavior).

File: src/components/calcite-action-pad/calcite-action-pad.e2e.ts [hunk 4]
```diff
@@ -1,18 +1,108 @@
...
+    await page.setContent('<calcite-action-pad expand="false"></calcite-action-pad>');
+
+    await page.waitForChanges();
+
+    const expandAction = await page.find("calcite-action-pad >>> calcite-action");
+
+    expect(expandAction).toBeNull();
+  });
+
+  it("expanded", async () => {
```
Comment (jcfranco) [inline src/components/calcite-action-pad/calcite-action-pad.e2e.ts]: Nitpick: should this be renamed to indicate that it's testing the toggling behavior?

File: src/components/calcite-action-pad/calcite-action-pad.e2e.ts [hunk 5]
```diff
@@ -1,18 +1,108 @@
...
+
+    expect(eventSpy).toHaveReceivedEvent();
+  });
+
+  it("expanded by default", async () => {
+    const page = await newE2EPage();
+
+    await page.setContent("<calcite-action-pad expanded></calcite-action-pad>");
+
+    const buttonGroup = await page.find("calcite-action-pad >>> .action-group--bottom");
```
Comment (jcfranco) [inline src/components/calcite-action-pad/calcite-action-pad.e2e.ts]: Can you use the CSS lookup to build the group selector?

File: src/components/calcite-action-pad/calcite-action-pad.e2e.ts [hunk 6]
```diff
@@ -1,18 +1,108 @@
...
+
+    await page.waitForChanges();
+
+    expect(eventSpy).toHaveReceivedEvent();
+  });
+
+  it("expanded by default", async () => {
+    const page = await newE2EPage();
+
+    await page.setContent("<calcite-action-pad expanded></calcite-action-pad>");
```
Comment (jcfranco) [inline src/components/calcite-action-pad/calcite-action-pad.e2e.ts]: Nitpick: the test name throws me off. It says *expanded by default*, but the attribute is set here. Is this needed?

File: src/components/calcite-action-pad/calcite-action-pad.e2e.ts [hunk 7]
```diff
@@ -1,18 +1,108 @@
...
+  });
+
+  it("expanded change should fire event", async () => {
+    const page = await newE2EPage();
+
+    await page.setContent("<calcite-action-pad></calcite-action-pad>");
+
+    const element = await page.find("calcite-action-pad");
+
+    const eventSpy = await page.spyOnEvent("calciteActionPadToggle", "window");
```
Comment (jcfranco) [inline src/components/calcite-action-pad/calcite-action-pad.e2e.ts]: Can you spy on the element itself?

File: src/components/calcite-action-pad/calcite-action-pad.e2e.ts [hunk 8]
```diff
@@ -16,94 +17,96 @@ describe("calcite-action-pad", () => {
...
+    });
   });
 
   it("should be accessible", async () =>
     accessible(`
     <calcite-action-pad>
-      <calcite-action-group>
-        <calcite-action text="Add">
-        </calcite-action>
+      (-action as calcite)-group>
```
Comment (jcfranco) [inline src/components/calcite-action-pad/calcite-action-pad.e2e.ts]: Can you fix this markup (same below)?

File: src/components/interfaces.ts [hunk 9]
```diff
@@ -8,6 +8,6 @@ export type CalciteBlockSectionToggleDisplay = "button" | "switch";
 
 export type FlowDirection = "advancing" | "retreating";
 
-export type CalciteActionAppearance = "solid" | "clear";
```
Comment (jcfranco) [inline src/components/interfaces.ts]: `calcite-action.tsx` still references this: https://github.com/Esri/calcite-app-components/blob/e1af1d60ce4698a2a41b72b5237d1ce4a862d772/src/components/calcite-action/calcite-action.tsx#L3

**Pull request merged**